src/toxcore/dht/server/mod.rs

```diff
@@ -266,35 +274,35 @@ impl Server {
         self.initial_bootstrap.push(pn);
     }
 
-    /// Run initial bootstapping, It continues to sending NodesRequest to bootstrap nodes
-    /// until at least one entry inserted to Kbucket
-    /// and send NodesRequest to nodes in Kbucket if all nodes are discarded.
-    /// All nodes in Kbucket are discarded means that network is not available for a while.
-    fn run_recover_network_error(self) -> IoFuture<()> {
+    /// Run initial bootstrapping. It sends `NodesRequest` packet to bootstrap
+    /// nodes periodically if all nodes in Kbucket are discarded (including the
+    /// case when it's empty). It has to be an endless loop because we might
+    /// loose the network connection and thereby loose all nodes in Kbucket.
+    fn run_bootstrap_requests_sending(self) -> IoFuture<()> {
```

Comment:
oh wait, why do we accept self by value?

Reply:
Because it's consumed inside. I believe if something is consumed it's supposed to be passed by value rather then by reference with copying inside.

Reply:
> Because it's consumed inside

why?

Reply:
Because it has to be moved inside endless loop: https://github.com/tox-rs/tox/blob/818c22a56442181276e62d887f822205f7ec9f16/src/toxcore/dht/server/mod.rs#L284
It can't be used by reference there. So either we take it by value or clone inside.

Reply:
Thanks